agent/command/git.go

```diff
@@ -810,7 +809,7 @@ func (c *gitFetchProject) fetch(ctx context.Context,
 		if err := ctx.Err(); err != nil {
 			return errors.Wrapf(err, "canceled while applying module '%s'", moduleName)
 		}
-		err = c.fetchModuleSource(ctx, conf, logger, jpm, opts.method, opts.token, p, moduleName)
+		err = c.fetchModuleSource(ctx, conf, logger, jpm, opts.token, p, moduleName)
```

Comment:
It looks like these were already passing in opts.token. Where does the fix happen for 
"they do not pass a token to Github in the clone command"?

Reply:
The fix is [here](https://github.com/evergreen-ci/evergreen/pull/6709/files#diff-1e8d12ae20a5922169a78aaae96aab11adb323b77baf8910c3cbfba80b1d0012R697), to make the clone method HTTPS if we encounter an HTTPS url. Beforehand if an HTTPS url is detected the default distro clone method would get set, which is often legacy SSH. This causes a problem because when the clone command is built, if the method is legacy SSH it will create a clone command that [does not use the token](fetchModuleSource)